app/helpers/application_helper.rb

```diff
@@ -15,4 +15,12 @@ def shorten_long(input)
     end
     return_string
   end
+
+  def csv_route(type)
+    if ENV['RAILS_RELATIVE_URL_ROOT'].nil?
+      "/csv/#{type}.csv"
+    else
+      ENV['RAILS_RELATIVE_URL_ROOT'] + "csv/#{type}.csv"
```

Comment:
in your else statement you do still need the "/" at the beginning of the string.  The variable will contain "/aaec" so the string will need to be "/csv/#{type}.csv"

Also a slicker way of doing this without a conditional is just to use `ENV['RAILS_RELATIVE_URL_ROOT'].to_s + "/csv/#{type}.csv"`.  If the variable is `nil` the `.to_s` will convert it to an empty string.